File: Chat/Server/Data/Entity/Space.cs
namespace Chat.Server.Data.Entity;

public class Space
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public string Description { get; set; } = string.Empty;

    public ICollection<SpaceMembership> Memberships { get; set; } = new HashSet<SpaceMembership>();

    public ICollection<SpaceMessage> Messages { get; set; } = new HashSet<SpaceMessage>();
    public DateTime Created { get; set; } = DateTime.UtcNow;
    public DateTime Updated { get; set; } = DateTime.UtcNow;
}

File: Chat/Client/web/mocks/models/space.ts
import { Message } from "./message"
import User from "./user"

export interface SpaceInfo {
    space: Space,
    messages: Message[],
    members: User[],
}

export interface Space {
    __typename: string,
    id: number,
    name: string,
    updated: string,
    description: string,
}